development/build/index.js

```diff
@@ -117,4 +133,77 @@ function defineAllTasks() {
 
   // special build for minimal CI testing
   createTask('styles', styleTasks.prod);
+
+  // Finally, start the build process by running the entry task.
+  runTask(entryTask, { skipStats });
+}
+
+function parseArgv() {
+  const NamedArgs = {
+    BetaVersion: 'beta-version',
+    BuildType: 'build-type',
+    OmitLockdown: 'omit-lockdown',
+    SkipStats: 'skip-stats',
+  };
+
+  const entryTask = process.argv[2];
+  if (!entryTask) {
+    throw new Error('MetaMask build: No entry task specified.');
+  }
+
+  // The entry task should never be prefixed with '-' or '--', so that is easily
+  // distinguishable from other arguments.
+  if (entryTask.startsWith('-')) {
+    throw new Error(`MetaMask build: invalid entry task: ${entryTask}`);
+  }
+
+  const argv = minimist(process.argv.slice(3), {
+    boolean: [NamedArgs.OmitLockdown, NamedArgs.SkipStats],
+    string: [NamedArgs.BuildType],
+    default: {
+      [NamedArgs.BetaVersion]: 0,
+      [NamedArgs.BuildType]: 'main',
+      [NamedArgs.OmitLockdown]: false,
+      [NamedArgs.SkipStats]: false,
+    },
+  });
+
+  const betaVersion = argv[NamedArgs.BetaVersion];
+  if (!Number.isInteger(betaVersion) || betaVersion < 0) {
+    throw new Error(`MetaMask build: Invalid beta version: ${betaVersion}`);
+  }
+
+  // Arguments that are used throughout or deeply within the build system are
+  // set as environment variables.
+  setImmutableProperties(process.env, [
```

Comment:
id rather not if we can just put it in a configuration blob and pass that around

Reply:
![image](https://user-images.githubusercontent.com/25517051/132589650-53a78a27-953e-41c7-827d-92d274f2d781.png)

Reply:
<img src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fmedia.tenor.co%2Fimages%2F3ee32071e0d202ac9ab147b1393491e1%2Fraw&f=1&nofb=1" />

Reply:
That's too much refactoring / tracing through the build system than I'd like to do right now, if you're OK with that. In addition, these environment variables are just replacing other environment variables we were already using. Now we just define them in a single spot.

Reply:
In other words, it's an improvement! 😄

Reply:
Alright! Fine! I did it! 604e82e